scripts/setup.sh

```diff
@@ -222,9 +206,55 @@ gcloud config set project $GCP_PROJECT
 UPDATE=$(gcloud functions list --filter=$GCP_FUNCTION_NAME --project="$GCP_PROJECT" --format="value(name)")
 INSTALL=true
 
-if [ -n  "$UPDATE" ]; then
-echo -e "- function \e[1;32m $GCP_FUNCTION_NAME \e[0m already exists in $GCP_PROJECT and it will be updated"
-INSTALL=false
+if [ -n "$UPDATE" ]; then
+  echo -e "- function \e[1;32m $GCP_FUNCTION_NAME \e[0m already exists in $GCP_PROJECT and it will be updated"
+  INSTALL=false
+fi
+
+# Create App Engine app if not exists
+echo
+echo "- checking App Engine app, required for Cloud Scheduler"
+APP_ENGINE=$(gcloud app describe --format="json" 2>/dev/null || true)
+SERVING_APP_ENGINE=$(echo "$APP_ENGINE" | jq -r '.servingStatus')
+
+if [[ -z "$APP_ENGINE" ]]; then
+  echo
+  echo "To continue deployment your GCP project must contain an App Engine app - it's required for Cloud Scheduler"
+  while true; do
+    read -p "Do you want to create App Engine app? [y/n]" yn
+    case $yn in
+    [Yy]*)
+      echo
+      echo "Please provide the region for App Engine app."
+      echo
+      echo "Available regions:"
+      APP_ENGINE_LOCATIONS=$(gcloud app regions list --format="json" | jq -r '.[] | .region')
+      echo "$APP_ENGINE_LOCATIONS"
+      readarray -t LOCATIONS_ARR <<<"$(echo "${APP_ENGINE_LOCATIONS}")"
+      echo
+      while ! [[ " ${LOCATIONS_ARR[*]} " == *" $APP_ENGINE_LOCATION "* ]]; do
+        read -p "Enter location for App Engine app: " -e APP_ENGINE_LOCATION
+      done
+      echo
+      echo "- creating the App Engine app"
+      gcloud app create -q --region="$APP_ENGINE_LOCATION"
+      break
+      ;;
+    [Nn]*)
+      echo
+      echo -e "\e[91mERROR: \e[37mCannot continue without App Engine. Deployment aborted."
+      exit
+      ;;
+    *) echo "- please answer y or n" ;;
+    esac
+  done
+elif [[ "$SERVING_APP_ENGINE" != "SERVING"  ]]; then
+  echo
+  echo -e "\e[91mERROR: \e[37mTo continue deployment your GCP project must contain an App Engine app - it's required for Cloud Scheduler"
+  echo
+  echo 'Enable App Engine on: https://console.cloud.google.com/appengine/settings?_ga=2.67541841.1390822090.1635945824-1034520987.1633950930'
```

Comment:
should this link be shorter?
https://console.cloud.google.com/appengine/settings